src/components/flat-table/flat-table.style.js

```diff
@@ -2,21 +2,49 @@ import styled, { css } from 'styled-components';
 import StyledFlatTableHeader from './flat-table-header/flat-table-header.style';
 import StyledFlatTableRowHeader from './flat-table-row-header/flat-table-row-header.style';
 import StyledFlatTableHead from './flat-table-head/flat-table-head.style';
+import { baseTheme } from '../../style/themes';
 
 const StyledFlatTableWrapper = styled.div`
   height: 100%;
+
+  ${({
+    colorTheme, styleOverride, theme
+  }) => {
+    switch (colorTheme) {
+      case 'dark':
+        return css`
+        ${StyledFlatTableHeader} {
+          background-color: ${theme.flatTable.dark.headerBackground};
+          border-right: 1px solid ${theme.flatTable.dark.border};
+          color: ${theme.colors.white};
+        }`;
+
+      case 'light':
+        return css`
+        ${StyledFlatTableHeader} {
+          background-color: ${theme.flatTable.light.headerBackground};  
+          border-right: 1px solid ${theme.flatTable.light.border};
+        }`;
+
+      default:
+        return css`
+        ${StyledFlatTableHeader} {
+          background-color: ${styleOverride || theme.flatTable.default.headerBackground};
+        }`;
+    }
+  }}
+
   ${({ hasStickyHead }) => hasStickyHead && css`
     overflow-y: auto;
 
     ${StyledFlatTableHeader} {
-      background-color: #fff;
       position: sticky;
       z-index: 1;
-    }
+    };
```

Comment:
Don't think `;` is needed

Reply:
fixed in https://github.com/Sage/carbon/pull/2866/commits/840d7cd4461c614101f121bf05a818cb440074d9